File: ebean-autotune/src/test/java/org/tests/model/basic/Order.java
package org.tests.model.basic;

import javax.persistence.*;
import java.time.LocalDate;

@Entity
@Table(name = "o_order")
public class Order extends BaseModel {

  public enum Status {
    NEW,
    APPROVED,
    SHIPPED,
    COMPLETE
  }


  @Enumerated(EnumType.STRING)
  Status status = Status.NEW;

  LocalDate orderDate;

  @ManyToOne(cascade = CascadeType.PERSIST)
  final Customer customer;

  public Order(Customer customer) {
    this.customer = customer;
  }

  public Status getStatus() {
    return status;
  }

  public void setStatus(Status status) {
    this.status = status;
  }

  public LocalDate getOrderDate() {
    return orderDate;
  }

  public void setOrderDate(LocalDate orderDate) {
    this.orderDate = orderDate;
  }

  public Customer getCustomer() {
    return customer;
  }

}

File: tests/test-kotlin/src/main/kotlin/org/example/Order.kt
package org.example.order

import jakarta.persistence.CascadeType
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import jakarta.persistence.Version

@Entity
@Table(name = "t_order")
class Order(
  customer: Customer,
) {
  @Id
  val id: Long = 0

  @ManyToOne
  val customer: Customer = customer

  @OneToMany(mappedBy = "order", cascade = [CascadeType.ALL], orphanRemoval = true)
  val items: MutableList<OrderItem> = mutableListOf()

  @Version
  val version: Long = 0
}
